make sure the download from a mirror is tried before checkout from VCS

diff --git a/include/download.mk b/include/download.mk
--- a/include/download.mk
+++ b/include/download.mk
@@ -159,7 +159,7 @@ define Download/Defaults
   MD5SUM:=
   SUBDIR:=
   MIRROR:=1
-  MIRROR_MD5SUM:=x
+  MIRROR_MD5SUM:=xxx
   VERSION:=
 endef
 
diff --git a/include/host-build.mk b/include/host-build.mk
--- a/include/host-build.mk
+++ b/include/host-build.mk
@@ -129,7 +129,7 @@ define Download/default
   SUBDIR:=$(PKG_SOURCE_SUBDIR)
   VERSION:=$(PKG_SOURCE_VERSION)
   MD5SUM:=$(PKG_MD5SUM)
-  MIRROR_MD5SUM:=$(PKG_MIRROR_MD5SUM)
+  $(if $(PKG_MIRROR_MD5SUM),MIRROR_MD5SUM:=$(PKG_MIRROR_MD5SUM))
 endef
 
 define Host/Exports/Default
